README.md

```diff
@@ -20,7 +20,7 @@ The available Go binaries
 |  `BUILDKITE_PARALLEL_JOB_COUNT` | - | Required, total number of parallelism |
 |  `BUILDKITE_PARALLEL_JOB` | - | Required, test plan for specific node |
 | `BUILDKITE_SPLITTER_SUITE_TOKEN` | `BUILDKITE_ANALYTICS_TOKEN` | Required, unique token for Test Suite that is being parallelised |
-|  `BUILDKITE_SPLITTER_IDENTIFIER` | `BUILDKITE_BUILD_ID` | Required, use default unless you have a specific use case |
+|  `BUILDKITE_SPLITTER_IDENTIFIER` | `BUILDKITE_BUILD_ID/BUILDKITE_STEP_ID` | Required, use default unless you have a specific use case |
```

Comment:
By making this default to a compound identifier based on the Build ID and Step ID, I don't think this is 'Required' any more as suggested by the description, so we perhaps need to wordsmith a better description for this one.

There are two things that stand out to me:
1. The use of the `BUILDKITE_SPLITTER_IDENTIFIER` can be used to override the identifier used by the test-splitter.
2. If the `BUILDKITE_SPLITTER_IDENTIFIER` environment variable is not overridden, the test splitter _must_ have access to the `BUILDKITE_BUILD_ID` and `BUILDKITE_STEP_ID` environment variables to create the compound identifier.

Reply:
This is a good point 💯. The identifier was never "required" by the client because it will set a default value, but the user has to pass the `BUILDKITE_BUILD_ID` and `BUILDKITE_STEP_ID`. We might also need to explain the usage, and how the test splitter use the identifier. 

Maybe something along these lines?
> Optional. Test Splitter use identifier to store and fetch the test plan and must be unique for each build and steps group. By default it will use a composite of `BUILDKITE_BUILD_ID` and `BUILDKITE_STEP_ID`, but it can be overriden by specifying the `BUILDKITE_SPLITTER_IDENTIFIER`. `BUILDKITE_BUILD_ID` and `BUILDKITE_STEP_ID` must be accessible by the client when using the default.

Reply:
Good catch! It makes sense to me that the identifier shouldn't be required. 
I've updated the description based on @nprizal's suggestion.

Reply:
Yeah looks good :+1: